test/e2e/deploy/volumeleaks.go

```diff
@@ -44,15 +45,29 @@ func GetHostVolumes(d *Deployment) map[string][]string {
 	// script appears to be "no such file".
 	for worker := 1; ; worker++ {
 		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
-		ssh := exec.Command(sshcmd, cmd)
-		// Intentional Output instead of CombinedOutput to dismiss warnings from stderr.
-		// lvs may emit lvmetad-related WARNING msg which can't be silenced using -q option.
-		out, err := ssh.Output()
-		if err != nil && os.IsNotExist(err) {
+		if _, err := os.Stat(sshcmd); err == nil {
+			i := 0
+			for {
+				ssh := exec.Command(sshcmd, cmd)
+				// Intentional Output instead of CombinedOutput to dismiss warnings from stderr.
+				// lvs may emit lvmetad-related WARNING msg which can't be silenced using -q option.
+				out, err := ssh.Output()
+				if err != nil {
+					if i >= 3 {
+						ginkgo.Fail(fmt.Sprintf("repeated ssh attempts to node %d failed", worker))
```

Comment:
The error is still not part of the failure message.

Reply:
showing error when all attempts have failed, seems layering violation for me. We show error each time ssh fails. After 3 attempts failed, the problem is not one of the 3 errors, but aggregate state "all attempts failed". It's not guaranteed errors were same, so which one should we show at the end, last one? (right, we dont store others so thats the only one we have). But all errors were already shown. You want to repeat the last one? But why it is special that it needs to be shown twice while it happened once?

Reply:
When I see the test failure, I want to know immediately why ssh didn't work. It's faster if I don't have to go hunting for the error somewhere in the log file. Reporting the error directly in the failure message achieves that.

If I then want to double-check that all previous attempts failed the same way, then I can still consult the log file.

Reply:
I added error print now in final msg. Actually, my logic about "aggregated" report and repeated log does not hold water 100%, because in current code, we check for repeat limit before reporting error, thus the last report is not repeated.